### ISSUE TITLE
forum spammed today - look screen

### ISSUE THREAD

**Issue by UnrealCatze** [PHPFusion/PHPFusion#1182]
new register and he have post around 20 threads with nearly same texts

![unbenannt](https://cloud.githubusercontent.com/assets/22166864/20837215/5ef57664-b8a3-11e6-9769-b9b1f26ab401.JPG)

**Comment by UnrealCatze** [comment]
look about this email jokergreen0220@yahoo.com and you will find many more sites where be spamed and hacked

**Comment by JoakimFalk** [comment]
I would not want to call it "hack", where is the hack?. All I see is a persistent spammer.
Chan will make some amends to strikt token and to add captcha on flood control I think, check your flood control settings as well.

**Comment by meangczac** [comment]
Security over performance. Token recycling will be removed. Wait for git.

I will check form url for token.
I will check javascript based injection.
All request will be sanitized as well. 

Flood please make it at least MINUMUM 30-45 seconds. 

I will remove the benefits of spamming site. All links will default to deter site crawlers to nofollow. 
I will add this into defender and stripinput function.

**Comment by meangczac** [comment]
I checked the token security layer, and beside to make it more secure, I can only think of static variables so it will not be affected by mutator methods. I reassigned the 4 original functions into a specific token  class file. The advantage is as following:
1. private variables unaffected by defender.
2. There is no mutator method in the most important variable and not even defender can access it - i.e. (string) $isValidToken
3. Sniff token function is deprecated. It now loads together with defender. No implementation. No documentation needed. It's automated
4. Second advantage of point 3. is that when defender is first loaded, no POST can be accepted. 

This is the increment. https://github.com/php-fusion/PHP-Fusion/commit/0ed85aca8c1b173b93c60a6b1d758501ca8f2157

Test case to prove the token's functioning...

testdefender.php to see the CSRF test. Your users keep posting manually... there is nothing we can do except increase to 45secs - 60seconds before allowing your users to post again.

Future...
------------------

https://akismet.com/plans/

WP charge people to subscribe to their akismet. Maybe we can provide the same service to our users too to provide a security IP dblayer checks. And then we can pay moderators to help maintain and moderate the IPs, research and handle troubleshooting etc.

If it works by all measures we can do on our code, there's no need for akismet.
